source/common/http/http1/codec_impl.cc

```diff
@@ -555,6 +555,24 @@ Http::Status ConnectionImpl::dispatch(Buffer::Instance& data) {
       [&](Buffer::Instance& data) -> Http::Status { return innerDispatch(data); }, data);
 }
 
+Http::Status ClientConnectionImpl::dispatch(Buffer::Instance& data) {
+  // TODO(#10878): Remove this wrapper when exception removal is complete. innerDispatch may either
+  // throw an exception or return an error status. The utility wrapper catches exceptions and
+  // converts them to error statuses.
+  return Utility::exceptionToStatus(
+      [&](Buffer::Instance& data) -> Http::Status {
+        Http::Status status = innerDispatch(data);
+        if (status.ok() && data.length() > 0) {
+          // The HTTP/1.1 codec pauses dispatch after a single response is complete. Extraneous data
+          // after a response is complete indicates an error.
+          return codecProtocolError(
+              "http/1.1 protocol error: extraneous data after response complete");
+        }
+        return status;
+      },
+      data);
+}
```

Comment:
You could call into the base class instead of reimplemeting exceptionToStatus and inner dispach here.

Http::Status status = ConnectionImpl::dispatch(data);
if (status.ok() && data.length() > 0) {
  // The HTTP/1.1 codec pauses dispatch after a single response is complete. Extraneous data
  // after a response is complete indicates an error.
  return codecProtocolError(
    "http/1.1 protocol error: extraneous data after response complete");
  }
return status;